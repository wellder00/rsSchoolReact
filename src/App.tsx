import { ErrorBoundary } from 'react-error-boundary';

import './index.scss';
import { Home } from './views/Home';
import { ErrorFallback } from './components/ErrorFallback';

function App() {
  return (
    <ErrorBoundary FallbackComponent={ErrorFallback}>
      <Home />;
    </ErrorBoundary>
  );
}

export default App;
