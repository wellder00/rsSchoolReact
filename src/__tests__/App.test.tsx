import { render, screen } from '@testing-library/react';
import App from '../App';

test('renders App component', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByAltText(/teleportBottom/i)).toBeInTheDocument();
});
