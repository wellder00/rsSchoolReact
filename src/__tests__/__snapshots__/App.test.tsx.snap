// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders App component 1`] = `
<DocumentFragment>
  <div
    class="_errorWrap_8c5621"
  >
    <h1
      class="_title_8c5621"
    >
      Oops!
    </h1>
    <p>
      Sorry, an unexpected error has occurred.
    </p>
    <p>
      Error
    </p>
  </div>
</DocumentFragment>
`;
